client/client.go

```diff
@@ -40,29 +40,21 @@ type DataExportClient interface {
 type Client struct {
 	HTTPClient *http.Client
 	Config     *config.Config
+	AuthFunc   func(*http.Request)
 }
 
 var _ DataExportClient = (*Client)(nil)
 
-// NewClient returns a Client initialized with http.DefaultClient and the
-// supplied apiToken.
-func NewClient(config *config.Config) *Client {
-	return &Client{
-		HTTPClient: http.DefaultClient,
-		Config:     config,
-	}
+func (c *Client) AddAuthHeaders(req *http.Request) {
+	c.AuthFunc(req)
 }
 
 // doReq performs the supplied HTTP request and returns the data in the response.
 // Necessary authentication headers are added before performing the request.
 //
 // If the error is nil, the caller is responsible for closing the returned data.
 func (c *Client) doReq(req *http.Request) (io.ReadCloser, error) {
-	req.Header.Set("Authorization", "Basic "+c.Config.FsApiToken)
```

Comment:
1) This code was removed, not moved. So this totally breaks this repo, right?
2) It could just move into a wrapper/decorator for `HTTPClient`, which then allows you to never need to call any auth func.

Reply:
When I did the reorg that is about to be posted here, I put this in the proper place (the auth func).  Please advise if this is sufficient.

Reply:
I think you should just remove `authFunc` entirely. The auth details can be left to the transport. See #92.

Reply:
Yep, did that.